File: WordGames/vowels/vowels.js
// Program to find all the words with no A's or E's and are at least 15 letters long
// Importing the fs module
let fs = require("fs")

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let words = readFileLines('/home/evergreen/UnderDogsAssignments/sowpods');
vowels = []
let aeiou = []
let aeiou_Alph = []

for(let i = 0; i < words.length; i++) { 
    if (words[i].includes("A") && words[i].includes("E") && words[i].includes("I") && words[i].includes("O") && words[i].includes("U")) {
        aeiou.push(words[i])}}

console.log("All the words all the vowels are" + aeiou)

for(let j = 0; j < aeiou.length; j++) {
    let a = (aeiou[j].indexOf('A'));
    let e = (aeiou[j].indexOf('E'));
    let i = (aeiou[j].indexOf('I'));
    let o = (aeiou[j].indexOf('O'));
    let u = (aeiou[j].indexOf('U'));
    if (a < e) {
        if (e < i) {
            if (i < o) {
                if (o < u) {
                    aeiou_Alph.push(aeiou[j]);
                }
            }
        }
    }
}
console.log("All words containing the vowels alphabetically are " + aeiou_Alph)

File: WordGames/vowels/vowels.py
# Program to create a list for all of the words with all the vowels
#currently getting hte wrong shortest word with all vowels
# importing the word list
sow_pods = open("/home/evergreen/UnderDogsAssignments/sowpods")

# Go through each word in the file and add it to the list, words
words = []
vowels = []

for line in sow_pods:
    words.append(line.strip())

for w in words:
    if ("A" in w) and ("E" in w) and ("I" in w) and ("O" in w)and ("U" in w):
        vowels.append(w)

print('All of the words that have  vowels, including Y are ' + str(vowels))

# Find the longest word that has no vowels
shortest = 'RHEUMATOIDALLY'
length = len(shortest)
for y in vowels:
    if len(y) < length:
        shortest = y
        length = len(shortest)

print('The shortest word with all the vowels in it is ' + str(shortest))

a = int()
e = int()
i = int()
o = int()
u = int()
vowels_alph = []

#Cycle through each index of vowels, find the placement of each vowel and then compare and 
# add to vowels_alph accordingly
for x in vowels:
    a = x.index("A")
    e = x.index("E")
    i = x.index("I")
    o = x.index("O")
    u = x.index("U")
    if a < e:
        if e < i:
            if i < o:
                #if o < u:
                    vowels_alph.append(x)
        

print(vowels_alph)